src/components/app/index.js

```diff
@@ -22,29 +22,25 @@ class App extends React.Component {
 
   componentDidUpdate(prevProps) {
     if (this.dataWasUpdated(prevProps.data, this.props.data)) {
-      this.store.dispatch(resetSnapshotData(formatSnapshots(this.props.data)));
+      this.resetStoreData(formatData(this.props.data));
     }
   }
 
   /**
-   * Quickly determine whether snapshots have been updated
+   * Quickly determine whether the pipeline data has been updated
    * @param {Object} prevData Previous data prop
    * @param {Object} newData New data prop
    */
   dataWasUpdated(prevData, newData) {
-    // Check just the schema IDs of incoming data updates
-    const dataID = ({ snapshots }) =>
-      Array.isArray(snapshots) && snapshots.map(d => d.schema_id).join('');
-
-    return dataID(prevData) !== dataID(newData);
+    return prevData.schema_id !== newData.schema_id;
```

Comment:
Do you plan this method to be used in the future? i.e. I can see it used once in componentDidUpdate - do you think it benefits from separating out the check into it's own function?

Reply:
I don't really mind either way - I had it as its own function previously because it was verbose check. Now it's simple. Could collapse it to make it more succinct, but this is easier to  understand, so I don't feel strongly either way.